nsv13/code/modules/munitions/ship_weapons/_ship_weapon.dm

```diff
@@ -0,0 +1,470 @@
+	var/fire_animation_length = 5
+	var/fire_mode
+
+	var/malfunction_sound = 'sound/effects/alert.ogg'
+
+	//Various traits that probably won't change
+	var/maintainable = TRUE //Does the weapon require maintenance?
+	var/bang = TRUE //Is firing loud?
+	var/auto_load = FALSE //Does the weapon feed and chamber the round once we load it?
+	var/semi_auto = FALSE //Does the weapon re-chamber for us after firing?
+
+	var/ammo_type = null
+	var/magazine_type = null
+	var/max_ammo = 1
+
+	var/datum/ship_weapon/weapon_type
+
+	// Things that change while we're operating
+	var/maint_req = 0 //Number of times a weapon can fire until a maintenance cycle is required. This will countdown to 0.
+	var/malfunction = FALSE
+	var/maint_state = MSTATE_CLOSED
+	var/safety = TRUE //Can only fire when safeties are off
+	var/loading = FALSE
+	var/state = STATE_NOTLOADED
+
+	var/obj/item/ammo_box/magazine/magazine //Magazine if we have one
+	var/obj/chambered //Chambered round if we have one. Extrapolate ammo type from this
+	var/list/ammo = list() //All loaded ammo
+
+/**
+ * Constructor for /obj/machinery/ship_weapon
+
+ * Attempts to link the weapon to an overmap ship.
+ * If the weapon requires maintenance, generates initial maintenance countdown.
+ * Caches icon state list for sanity checking when updating icons.
+ */
+/obj/machinery/ship_weapon/Initialize()
+	. = ..()
+	get_ship()
+	if(maintainable)
+		maint_req = rand(15,25) //Setting initial number of cycles until maintenance is required
+		create_reagents(50)
+	icon_state_list = icon_states(icon)
+
+/**
+ * Destructor for /obj/machinery/ship_weapon
+ * Try to unlink from a munitions computer, so it can re-link to other things
+ */
+/obj/machinery/ship_weapon/Destroy()
+	. = ..()
+	if(linked_computer)
+		linked_computer.SW = null
+
+/**
+ * Tries to link the ship to an overmap by finding the overmap linked it the area we are in.
+ */
+/obj/machinery/ship_weapon/proc/get_ship()
+	var/area/AR = get_area(src)
+	if(AR.linked_overmap)
+		linked = AR.linked_overmap
+		set_position(linked)
+
+/**
+ * Adds the weapon to the overmap ship's list of weapons of this type
+ */
+/obj/machinery/ship_weapon/proc/set_position(obj/structure/overmap/OM) //Use this to tell your ship what weapon category this belongs in
+	OM.add_weapon(src)
+
+/**
+ * If we're not already linked to an overmap ship, try again.
+ * If we can accept it as ammunition, try to load it.
+ * If we're in maintenance and it holds reagents, try to use it as lubricant.
+ */
+/obj/machinery/ship_weapon/attackby(obj/item/I, mob/user)
+	if(!linked)
+		get_ship()
+
+	if(ammo_type && istype(I, ammo_type))
+		load(I, user)
+		return
+	else if(magazine_type && istype(I, magazine_type))
+		load_magazine(I, user)
+		return
+	else if(istype(I, /obj/item/reagent_containers))
+		oil(I, user)
+		return
+	..()
+
+/**
+ * Unload magazine or just-loaded rounds.
+ */
+/obj/machinery/ship_weapon/pdc_mount/attack_hand(mob/user)
```

Comment:
why is the pdc_mount attack_hand in the ship_weapon file?

Reply:
Because I messed up